irohad/ametsuchi/impl/postgres_command_executor.cpp

```diff
@@ -1494,7 +1251,348 @@ namespace iroha {
+
+      PreparedStatement create_account{
+          "createAccount",
+          createAccountBase,
+          {(boost::format(R"(
+            has_perm AS (%s),)")
+            % checkAccountRolePermission(
+                  shared_model::interface::permissions::Role::kCreateAccount,
+                  "$1"))
+               .str(),
+           R"(AND (SELECT * FROM has_perm))",
+           R"(WHEN NOT (SELECT * FROM has_perm) THEN 1)"}};
+
+      prepareStatement(sql, create_account);
+
+      PreparedStatement create_asset{
+          "createAsset",
+          createAssetBase,
+          {(boost::format(R"(
+              has_perm AS (%s),)")
+            % checkAccountRolePermission(
+                  shared_model::interface::permissions::Role::kCreateAsset,
+                  "$1"))
+               .str(),
+           R"(WHERE (SELECT * FROM has_perm))",
+           R"(WHEN NOT (SELECT * FROM has_perm) THEN 1)"}};
+
+      prepareStatement(sql, create_asset);
+
+      PreparedStatement create_domain{
+          "createDomain",
+          createDomainBase,
+          {(boost::format(R"(
+              has_perm AS (%s),)")
+            % checkAccountRolePermission(
+                  shared_model::interface::permissions::Role::kCreateDomain,
+                  "$1"))
+               .str(),
+           R"(WHERE (SELECT * FROM has_perm))",
+           R"(WHEN NOT (SELECT * FROM has_perm) THEN 1)"}};
+
+      prepareStatement(sql, create_domain);
+
+      PreparedStatement create_role{
+          "createRole",
+          createRoleBase,
+          {(boost::format(R"(
+          account_has_role_permissions AS (
+                SELECT COALESCE(bit_or(rp.permission), '0'::bit(%s)) &
+                    $3 = $3
+                FROM role_has_permissions AS rp
+                JOIN account_has_roles AS ar on ar.role_id = rp.role_id
+                WHERE ar.account_id = $1),
+          has_perm AS (%s),)")
+            % std::to_string(bits)
+            % checkAccountRolePermission(
+                  shared_model::interface::permissions::Role::kCreateRole,
+                  "$1"))
+               .str(),
+           R"(WHERE (SELECT * FROM account_has_role_permissions)
+                          AND (SELECT * FROM has_perm))",
+           R"(WHEN NOT (SELECT * FROM
+                               account_has_role_permissions) THEN 2
+                        WHEN NOT (SELECT * FROM has_perm) THEN 3)"}};
+
+      prepareStatement(sql, create_role);
+
+      PreparedStatement detach_role{
+          "detachRole",
+          detachRoleBase,
+          {(boost::format(R"(
+            has_perm AS (%s),)")
+            % checkAccountRolePermission(
+                  shared_model::interface::permissions::Role::kDetachRole,
+                  "$1"))
+               .str(),
+           R"(AND (SELECT * FROM has_perm))",
+           R"(WHEN NOT (SELECT * FROM has_perm) THEN 1)"}};
+
+      prepareStatement(sql, detach_role);
```

Comment:
Maybe create a vector and do emplace_back instead of declaring variables? A loop can be used to prepare all the statements then.